h5grove/encoders.py

```diff
@@ -9,14 +8,15 @@
 from .utils import sanitize_array
 
 
-def bin_encode(array: Sequence[Number]) -> bytes:
-    """Sanitize an array and convert it to bytes.
+def bin_encode(array: np.ndarray, sanitize: bool = True) -> bytes:
```

Comment:
Add a sanitize arg to disable it when dtype is provided... but we might want to change this for consistency with the other formats?